### ISSUE TITLE
elasticsearch output 400 err

### ISSUE THREAD

**Issue by pierce300** [mozilla-services/heka#1630]
HI
   i found elasticsearch output that hava 400 bad request ,below is error information
HTTP response error status: 400 Bad Request "" {"error":"ElasticsearchParseException[Failed to derive xcontent from org.elasticsearch.common.bytes.BytesArray@1]","status":400}

i found http post data is null,

please help

**Comment by hobofan** [comment]
I also see some 400s in my log files:

```
2015/08/24 07:12:42 Plugin 'ElasticSearchOutput' error: HTTP response error status: 400 Bad Request
2015/08/24 08:11:18 Plugin 'ElasticSearchOutput' error: HTTP response error status: 400 Bad Request
```

After a few of those the output seems to go into some kind of failure state and won't deliver any more messages, causing my output_queue to fill up.

I am running 0.10.0b0

**Comment by simonpasquier** [comment]
I'm having similar issues with the queue filling up and never resuming. I'll investigate and try to find a way to reproduce it.

**Comment by simonpasquier** [comment]
@hobofan the issue with the output queue getting full has gone away for me with the the latest versions/0.10 branch. AFAICT this it is related to https://github.com/mozilla-services/heka/issues/1627 and fixed by the PR https://github.com/mozilla-services/heka/pull/1682.

I can't tell yet if the 400 errors still exist though.

**Comment by ioc32** [comment]
I still see a few of these 400s. I'll build versions/0.10 again and see if the issue still persists.

**Comment by sathieu** [comment]
@loc32: Apply #1744 and see what ES is complaining about.

**Comment by guluglu** [comment]
I also found this question in v0.10.0.   increase flush_interval  in elasticsearchoutput can aviod it or decrease error(it depend to flush_interval). but i dont know reason

**Comment by guluglu** [comment]
#1783

**Comment by rafrombrc** [comment]
Closed in #1795.
